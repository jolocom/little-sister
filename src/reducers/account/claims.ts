import { AnyAction } from 'redux'
import {
  CategorizedClaims,
  ClaimsState,
  DecoratedClaims,
} from 'src/reducers/account'
import { HAS_EXTERNAL_CREDENTIALS } from '../../actions/account/actionTypes'
import { CredentialTypes } from '../../lib/categories'

const categorizedClaims: CategorizedClaims = {
  Personal: [
    {
      credentialType: CredentialTypes.Name,
      claimData: {
        givenName: '',
        familyName: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
    },
  ],
  Contact: [
    {
      credentialType: CredentialTypes.Email,
      claimData: {
        email: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
      keyboardType: 'email-address',
    },
    {
      credentialType: CredentialTypes.MobilePhone,
      claimData: {
        telephone: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
      keyboardType: 'phone-pad',
    },
    {
      credentialType: CredentialTypes.PostalAddress,
      claimData: {
        addressLine1: '',
        addressLine2: '',
        postalCode: '',
        city: '',
        country: '',
      },
      id: '',
      issuer: {
        did: '',
      },
      subject: '',
    },
  ],
  // /** @dev FOR TESTING */
  Other: [],
}

export const initialState: ClaimsState = {
  selected: {
    credentialType: '',
    claimData: {},
    id: '',
    issuer: {
      did: '',
    },
    subject: '',
  },
  pendingExternal: {
    offeror: {
      did: '',
    },
    offer: [],
  },
  decoratedCredentials: categorizedClaims,
  hasExternalCredentials: false,
}

export const claims = (
  state = initialState,
  action: AnyAction,
): ClaimsState => {
  switch (action.type) {
    case 'SET_CLAIMS_FOR_DID':
      return { ...state, decoratedCredentials: addDefaultValues(action.claims) }
    case HAS_EXTERNAL_CREDENTIALS:
      return { ...state, hasExternalCredentials: action.value }
    case 'SET_EXTERNAL':
      return { ...state, pendingExternal: action.value }
    case 'RESET_EXTERNAL':
      return { ...state, pendingExternal: initialState.pendingExternal } // TODO Remove in favor of calling set external with empty array
    case 'SET_SELECTED':
      return { ...state, selected: action.selected }
    case 'RESET_SELECTED':
      return { ...state, selected: initialState.selected }
    case 'HANDLE_CLAIM_INPUT':
      // NOTE: this is handled slightly differently so that we prevent the
      // claimData keys from being re-ordered, as they are used to generate the
      // fields in ui/home/components/claimDetails
      const claimData = { ...state.selected.claimData }
      claimData[action.fieldName] = action.fieldValue

      return {
        ...state,
        selected: {
          ...state.selected,
          claimData,
        },
      }
    default:
      return state
  }
}

const addDefaultValues = (claims: CategorizedClaims) =>
  Object.keys(categorizedClaims).reduce(
    (acc: CategorizedClaims, category: string) => ({
      ...acc,
      [category]: injectPlaceholdersIfNeeded(category, claims[category]),
    }),
    {},
  )

const injectPlaceholdersIfNeeded = (
  category: string,
  claims: DecoratedClaims[],
): DecoratedClaims[] => {
  if (!claims || claims.length === 0) {
    return categorizedClaims[category]
  }

  const missing = categorizedClaims[category].filter(
    defaultClaim =>
      !claims.some(
        claim => claim.credentialType === defaultClaim.credentialType,
      ),
  )

  if (missing) {
    return [...missing, ...claims]
  }

  return claims
}
