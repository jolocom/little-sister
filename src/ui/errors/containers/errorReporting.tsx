import React, { useState } from 'react'
import { connect } from 'react-redux'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { AppError, reportError } from '../../../lib/errors'
import { ThunkDispatch } from '../../../store'
import { routeList } from '../../../routeList'
import { navigationActions } from '../../../actions'
import { Emoji, EmojiSection } from '../components/emojiSection'
import { JolocomButton, Wrapper } from '../../structure'
import { ScrollView } from 'react-native'
import { ChooseIssueSection } from '../components/chooseIssueSection'
import { DescriptionSection } from '../components/descriptionSection'
import { ContactSection } from '../components/contactSection'
import I18n from '../../../locales/i18n'
import strings from '../../../locales/strings'
import { SectionWrapper } from '../components/sectionWrapper'
import { NavigationSection } from '../components/navigationSection'

interface PaymentNavigationParams {
  error?: AppError | Error
  previousScreen: routeList
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, PaymentNavigationParams>
}

export enum Inputs {
  None,
  Dropdown,
  Description,
  Contact,
}

const ErrorReportingContainer = (props: Props) => {
  const { navigateToScreen, navigation } = props

  const [pickedIssue, setIssue] = useState('')
  const [description, setDescription] = useState('')
  const [contact, setContact] = useState('')
  const [currentInput, setInput] = useState(Inputs.None)
  const [toggleState, setToggle] = useState(false)
  const [selectedEmoji, setEmoji] = useState(Emoji.Empty)

  // NOTE send previous screen as props, since not all
  // error reports will have error objects

  // const { previousScreen } = navigation.state.params
  // const isBackButton = previousScreen === routeList.Settings
  const isBackButton = false

  // NOTE error reports can show up without an error,
  // hence a navigation prop (navigateTo) would be better than in the error object
  const navigateBack = () => {
    const error = props.navigation?.state?.params?.error
    const previousScreen = props.navigation?.state.params?.previousScreen
    if (error instanceof AppError) {
      navigateToScreen(error.navigateTo)
    } else {
      navigateToScreen(previousScreen || routeList.AppInit)
    }
  }

  const onSubmitReport = () => {
    const error = props.navigation?.state?.params?.error

    const userReport = {
      userError: pickedIssue,
      userDescription: description,
      userContact: contact,
      sendPrivateData: toggleState,
    }

    if (navigation) {
      reportError({ ...userReport, error })
      navigateBack()
    }
  }

  return (
    <Wrapper secondaryDark>
      <NavigationSection
        onNavigation={navigateBack}
        isBackButton={isBackButton}
      />
      <ScrollView overScrollMode="never">
        <SectionWrapper
          title={I18n.t(strings.TELL_US_THE_PROBLEM)}
          style={{ marginTop: 14 }}>
          <ChooseIssueSection
            currentInput={currentInput}
            pickedIssue={pickedIssue}
            onIssuePick={setIssue}
            setInput={setInput}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.CAN_YOU_BE_MORE_SPECIFIC)}>
          <DescriptionSection
            currentInput={currentInput}
            setInput={setInput}
            setDescription={setDescription}
            toggleState={toggleState}
            setToggle={setToggle}
            description={description}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.WANT_TO_GET_IN_TOUCH)}>
          <ContactSection
            onContactInput={setContact}
            currentInput={currentInput}
            setInput={setInput}
            contactValue={contact}
          />
        </SectionWrapper>
        <SectionWrapper title={I18n.t(strings.RATE_THE_ISSUE)}>
          <EmojiSection selectedEmoji={selectedEmoji} setEmoji={setEmoji} />
        </SectionWrapper>
        <JolocomButton
          onPress={onSubmitReport}
          containerStyle={{
            marginTop: 45,
            marginBottom: 66,
            marginHorizontal: 20,
            height: 56,
          }}
          text={I18n.t(strings.SUBMIT_REPORT)}
        />
      </ScrollView>
    </Wrapper>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  navigateToScreen: (screen: routeList) =>
    dispatch(navigationActions.navigate({ routeName: screen })),
})

export const ErrorReporting = connect(
  null,
  mapDispatchToProps,
)(ErrorReportingContainer)
