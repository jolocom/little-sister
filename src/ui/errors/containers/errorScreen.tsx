import React, { useEffect } from 'react'
import { BackHandler, StyleSheet, Text, View } from 'react-native'
import { JolocomButton, Wrapper } from '../../structure'
import { NavigationScreenProp, NavigationState } from 'react-navigation'
import { centeredText, fontMain, fontMedium } from '../../../styles/typography'
import { Colors, Typefaces } from '../../../styles'
import { BlueCircleImage, RedCircleImage } from '../../../resources'
import { BP } from '../../../styles/breakpoints'
import { connect } from 'react-redux'
import { ThunkDispatch } from '../../../store'
import { navigationActions } from '../../../actions'
import { routeList } from '../../../routeList'

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: Colors.overflowBlack,
    justifyContent: 'space-between',
  },
  imageWrapper: {
    marginTop: '15%',
    width: '65%',
    height: '54%',
  },
  title: {
    ...Typefaces.title2,
    ...centeredText,
    fontFamily: fontMedium,
    color: Colors.sandLight090,
    paddingHorizontal: 10,
  },
  message: {
    ...centeredText,
    fontFamily: fontMain,
    lineHeight: BP({
      small: 20,
      medium: 20,
      large: 24,
    }),
    fontSize: BP({
      small: 18,
      medium: 18,
      large: 20,
    }),
    marginTop: 13,
    paddingHorizontal: 26,
    color: Colors.sandLight080,
  },
  interactButton: {
    marginTop: 14,
    borderColor: 'white',
    borderWidth: 0.5,
  },
  buttonSection: {
    flex: 0.3,
    width: '100%',
    paddingHorizontal: 20,
    justifyContent: 'center',
  },
  contentSection: {
    flex: 0.7,
    width: '100%',
    alignItems: 'center',
  },
})

export enum ImageType {
  Red = 'red',
  Blue = 'blue',
}

export interface ErrorScreenParams {
  title: string
  message: string
  image?: ImageType
  interact?: {
    label: string
    onInteract: () => void | Promise<void>
  }
  dismiss: {
    label: string
    onDismiss: () => void | Promise<void>
  }
}

interface Props extends ReturnType<typeof mapDispatchToProps> {
  navigation: NavigationScreenProp<NavigationState, ErrorScreenParams>
}

const ErrorScreenContainer: React.FC<Props> = props => {
  const {
    resetNavigation,
    navigation: {
      state: {
        params: { title, message, interact, dismiss, image },
      },
    },
  } = props

  useEffect(() => {
    BackHandler.addEventListener('hardwareBackPress', onBackButtonPress)

    return () => {
      BackHandler.removeEventListener('hardwareBackPress', onBackButtonPress)
    }
  }, [])

  const onBackButtonPress = (): boolean => {
    resetNavigation()
    return true
  }

  return (
    <Wrapper style={styles.wrapper}>
      <View style={styles.contentSection}>
        <View style={styles.imageWrapper}>
          {image === ImageType.Red ? <RedCircleImage /> : <BlueCircleImage />}
        </View>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.message}>{message}</Text>
      </View>
      <View style={styles.buttonSection}>
        <JolocomButton
          onPress={dismiss.onDismiss}
          transparent
          text={dismiss.label}
        />
        {interact && (
          <JolocomButton
            containerStyle={styles.interactButton}
            transparent
            onPress={interact.onInteract}
            text={interact.label}
          />
        )}
      </View>
    </Wrapper>
  )
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  resetNavigation: () =>
    dispatch(navigationActions.navigate({ routeName: routeList.AppInit })),
})

export const ErrorScreen = connect(
  null,
  mapDispatchToProps,
)(ErrorScreenContainer)
