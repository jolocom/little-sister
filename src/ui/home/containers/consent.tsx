import React from 'react'
import { ConsentComponent } from 'src/ui/home/components/consent'
import { connect } from 'react-redux'
import { AnyAction } from 'redux'
import { RootState } from 'src/reducers/'

interface ConnectProps { }

interface Props extends ConnectProps {}

interface State {
  serviceProvider: object
  requestedClaims: string[] 
  checkboxList: object[]
}

//TODO: requestedClaims interface

export class ConsentContainer extends React.Component<Props, State> {

  state = {
    //serviceProvider should be passed in
    serviceProvider: { name: 'Jolocom', metadata: 'xyz'},
    //requestdClaims should be passed in
    requestedClaims: ['name', 'email', 'passport'],
    checkboxList: [],
  }

  componentDidMount() {
    this.mapRequestedClaimsToCheckboxes(this.state.requestedClaims)
  }

  private mapRequestedClaimsToCheckboxes(requestedClaims: string[]) {
    //typing
    const checkboxList: object[] = []
    requestedClaims.forEach((claim: string) => {
      checkboxList.push(
        {
          value: claim,
          label: claim,
          checked: false
        }
      )
    })
    //set into redux action here
    this.setState({ checkboxList })
  }

  private handleSubmitClaims() {
    console.log('submitting')
  }

  private handleDenySubmit() {
    console.log('denied')
  }

  render() {
    return (
    <ConsentComponent
      serviceProvider={ this.state.serviceProvider }
      checkboxList={ this.state.checkboxList }
      handleSubmitClaims={ this.handleSubmitClaims }
      handleDenySubmit={ this.handleDenySubmit }
     />
    )
  }
}

const mapStateToProps = (state: RootState) => {
  return {
  }
}

const mapDispatchToProps = (dispatch: (action: AnyAction) => void) => {
  return {
  }
}

export const Consent = connect(mapStateToProps, mapDispatchToProps)(ConsentContainer)