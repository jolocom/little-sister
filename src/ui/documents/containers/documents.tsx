import React from 'react'
import { connect } from 'react-redux'
import { Animated, ScrollView, StyleSheet, Text, View } from 'react-native'

import { openDocumentDetails } from 'src/actions/documents'
import { DecoratedClaims } from 'src/reducers/account'
import { RootState } from 'src/reducers'
import { getDocumentClaims } from 'src/utils/filterDocuments'
import { ThunkDispatch } from 'src/store'

import I18n from 'src/locales/i18n'
import { filters } from 'src/lib/filterDecoratedClaims'

import { DocumentsCarousel } from '../components/documentsCarousel'
import { DocumentsList } from '../components/documentsList'
import { DocumentViewToggle } from '../components/documentViewToggle'
import strings from '../../../locales/strings'
import { Typography, Colors } from 'src/styles'
import { Wrapper } from '../../structure'

interface Props
  extends ReturnType<typeof mapDispatchToProps>,
    ReturnType<typeof mapStateToProps> {}

interface State {
  activeDocumentIndex: number
  showingValid: boolean
}

const styles = StyleSheet.create({
  emptyDocumentsText: {
    ...Typography.mainText,
    ...Typography.centeredText,
    color: Colors.greyLight,
    paddingHorizontal: '5%',
  },
})

export class DocumentsContainer extends React.Component<Props, State> {
  public ScrollViewRef: ScrollView | null = null

  public state = {
    activeDocumentIndex: 0,
    showingValid: true,
  }

  private scrollToTop(): void {
    if (this.ScrollViewRef) {
      this.ScrollViewRef.scrollTo({ x: 0, y: 0, animated: true })
    }
  }

  private onActiveIndexChange(index: number): void {
    this.setState({ activeDocumentIndex: index })
    this.scrollToTop()
  }

  private handleToggle = () => {
    this.setState((prevState: State) => ({
      showingValid: !prevState.showingValid,
      activeDocumentIndex: 0,
    }))
    this.scrollToTop()
  }

  public render(): JSX.Element {
    const { openDocumentDetails, decoratedCredentials } = this.props
    const documents = getDocumentClaims(decoratedCredentials['Other'])
    const docFilter = this.state.showingValid
      ? filters.filterByValid
      : filters.filterByExpired
    const displayedDocs = docFilter(documents)
    const isEmpty = displayedDocs.length === 0
    const otherIsEmpty = displayedDocs.length === documents.length

    if (isEmpty) {
      return (
        <Wrapper centered>
          <Text style={styles.emptyDocumentsText}>
            {I18n.t(strings.NO_DOCUMENTS_TO_SEE_HERE) + '...'}
          </Text>
        </Wrapper>
      )
    }

    return (
      <Wrapper>
        <Animated.View style={{ flex: 1 }}>
          {!otherIsEmpty && (
            <DocumentViewToggle
              showingValid={this.state.showingValid}
              onTouch={this.handleToggle}
            />
          )}
          <ScrollView
            // to scroll to top upon changing card
            ref={ref => (this.ScrollViewRef = ref)}
            contentContainerStyle={
              isEmpty ? { flex: 1 } : { paddingBottom: 110 }
            }
            scrollEnabled={!isEmpty}
          >
            {this.state.showingValid ? (
              <DocumentsCarousel
                documents={displayedDocs}
                activeIndex={this.state.activeDocumentIndex}
                onActiveIndexChange={this.onActiveIndexChange.bind(this)}
              />
            ) : (
              <DocumentsList
                documents={displayedDocs}
                onDocumentPress={openDocumentDetails}
              />
            )}
          </ScrollView>
        </Animated.View>
      </Wrapper>
    )
  }
}

const mapStateToProps = ({
  account: {
    claims: { decoratedCredentials },
  },
}: RootState) => ({
  decoratedCredentials,
})

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  openDocumentDetails: (doc: DecoratedClaims) =>
    dispatch(openDocumentDetails(doc)),
})

export const Documents = connect(
  mapStateToProps,
  mapDispatchToProps,
)(DocumentsContainer)
