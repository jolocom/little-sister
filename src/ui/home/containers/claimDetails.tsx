import React from 'react'
import { ClaimDetailsComponent } from 'src/ui/home/components/claimDetails'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { accountActions } from 'src/actions'
import { ClaimsState } from 'src/reducers/account'
import Immutable from 'immutable'

interface ConnectProps {
  claims: ClaimsState
  saveClaim: () => void
  handleClaimInput: (fieldValue: string, fieldName: string) => void
}

interface Props extends ConnectProps {}

interface State {}

export class ClaimDetailsContainer extends React.Component<Props, State> {
  render() {
    return (
      <ClaimDetailsComponent
        saveClaim={this.props.saveClaim}
        handleClaimInput={this.props.handleClaimInput}
        selectedClaim={this.props.claims.selected}
      />
    )
  }
}

const mapStateToProps = (state: RootState) => {
  const claims = Immutable.fromJS(state.account.claims)
  return {
    claims: claims.toJS(),
  }
}

const mapDispatchToProps = (dispatch: Function) => ({
  saveClaim: () => {
    dispatch(accountActions.saveClaim())
  },
  handleClaimInput: (fieldValue: string, fieldName: string) => {
    dispatch(accountActions.handleClaimInput(fieldValue, fieldName))
  },
})

export const ClaimDetails = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClaimDetailsContainer)
