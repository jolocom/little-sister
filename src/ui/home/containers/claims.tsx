import React from 'react'
import { connect } from 'react-redux'
import { View } from 'react-native'

import { CredentialOverview } from '../components/credentialOverview'
import { accountActions, ssoActions, navigationActions } from 'src/actions'
import { ClaimsState } from 'src/reducers/account'
import { DecoratedClaims } from 'src/reducers/account/'
import { routeList } from 'src/routeList'

interface ConnectProps {
  setClaimsForDid: () => void
  toggleLoading: (val: boolean) => void
  openClaimDetails: (claim: DecoratedClaims) => void
  openScanner: () => void
  parseJWT: (jwt: string) => void
  did: string
  claims: ClaimsState
  loading: boolean
}

interface Props extends ConnectProps {}

export class ClaimsContainer extends React.Component<Props> {
  componentWillMount() {
    this.props.setClaimsForDid()
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <CredentialOverview
          did={this.props.did}
          claimsState={this.props.claims}
          loading={!!this.props.claims.loading}
          onEdit={this.props.openClaimDetails}
        />
      </View>
    )
  }
}

// TODO nicer pattern for accessing state, perhaps immer or something easier to Type
const mapStateToProps = (state: any) => ({
  did: state.account.did.toJS().did,
  claims: state.account.claims.toJS(),
  loading: state.account.loading.toJS().loading,
})

const mapDispatchToProps = (dispatch: Function) => ({
  parseJWT: (jwt: string) => dispatch(ssoActions.parseJWT(jwt)),
  openClaimDetails: (claim: DecoratedClaims) =>
    dispatch(accountActions.openClaimDetails(claim)),
  setClaimsForDid: () => dispatch(accountActions.setClaimsForDid()),
  toggleLoading: (val: boolean) => dispatch(accountActions.toggleLoading(val)),
  openScanner: () =>
    dispatch(
      navigationActions.navigate({ routeName: routeList.QRCodeScanner }),
    ),
})

export const Claims = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClaimsContainer)
