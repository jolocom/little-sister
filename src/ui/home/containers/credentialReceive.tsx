import React from 'react'
import { connect } from 'react-redux'
import { RootState } from 'src/reducers/'
import { DecoratedClaims } from 'src/reducers/account'

import Immutable from 'immutable'
import {
  convertToDecoratedClaim,
  saveExternalCredentials,
} from 'src/actions/account'
import { CredentialDialogComponent } from '../components/credentialDialog'
import { Block, Container } from 'src/ui/structure'
import { SignedCredential } from 'jolocom-lib/js/credentials/signedCredential/signedCredential'
import { cancelReceiving } from 'src/actions/sso'
import { ButtonSection } from 'src/ui/structure/buttonSectionBottom'

interface ConnectProps {
  externalCredentials: SignedCredential[]
  saveExternalCredentials: () => void
  goBack: () => void
}

interface Props extends ConnectProps {}

interface State {
  toRender: DecoratedClaims
}

export class CredentialsReceiveContainer extends React.Component<Props, State> {
  state = {
    toRender: convertToDecoratedClaim(this.props.externalCredentials)[0],
  }

  render() {
    return (
      <Container style={{ padding: 0, justifyContent: 'space-between' }}>
        <Block flex={0.9}>
          <CredentialDialogComponent credentialToRender={this.state.toRender} />
        </Block>
        <ButtonSection
          confirmText={'Accept'}
          denyText={'Deny'}
          handleConfirm={this.props.saveExternalCredentials}
          handleDeny={this.props.goBack}
          disabled={false}
        />
      </Container>
    )
  }
}

const mapStateToProps = (state: RootState) => {
  const claims = Immutable.fromJS(state.account.claims)
  return {
    externalCredentials: claims.toJS().pendingExternal,
  }
}

const mapDispatchToProps = (dispatch: (action: Function) => void) => ({
  saveExternalCredentials: () => dispatch(saveExternalCredentials()),
  goBack: () => dispatch(cancelReceiving()),
})

export const CredentialReceive = connect(
  mapStateToProps,
  mapDispatchToProps,
)(CredentialsReceiveContainer)
