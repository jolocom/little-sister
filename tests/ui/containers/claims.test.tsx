import React from 'react'
import { shallow } from 'enzyme'
import { ClaimsContainer } from 'src/ui/home/containers/claims'

describe('Claims container', () => {
  const COMMON_PROPS = {
    showClaimsDetails: false,
    typeClaimDetails: '',
    claims: {},
    toggleLoading: () => {},
  }

  it('mounts correctly and matches the snapshot', () => {
    const setClaimsForDid = jest.fn()

    const props = Object.assign({}, COMMON_PROPS, {
      setClaimsForDid,
    })

    const rendered = shallow(<ClaimsContainer {...props} />)

    expect(rendered).toMatchSnapshot()
    expect(setClaimsForDid).toHaveBeenCalledTimes(1)
  })

  it('correctly changes scanning to true when qr code scanner is started', () => {
    const props = Object.assign({}, COMMON_PROPS, {
      setClaimsForDid: () => {},
    })

    const rendered = shallow(<ClaimsContainer {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('correctly changes scanning to false when qr code scanner is canceled', () => {
    const onScannerCancel = jest.fn()

    const props = Object.assign({}, COMMON_PROPS, {
      setClaimsForDid: () => {},
    })

    const rendered = shallow(<ClaimsContainer {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
