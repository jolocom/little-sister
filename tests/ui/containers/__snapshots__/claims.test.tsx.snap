// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claims container correctly changes scanning to false when qr code scanner is canceled 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ClaimOverview
    claims={Object {}}
    onScannerStart={[Function]}
    openClaimDetails={[Function]}
    scanning={false}
  />
</Component>
`;

exports[`Claims container correctly changes scanning to true when qr code scanner is started 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ClaimOverview
    claims={Object {}}
    onScannerStart={[Function]}
    openClaimDetails={[Function]}
    scanning={false}
  />
</Component>
`;

exports[`Claims container mounts correctly and matches the snapshot 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ClaimOverview
    claims={Object {}}
    onScannerStart={[Function]}
    openClaimDetails={[Function]}
    scanning={false}
  />
</Component>
`;
